### PR TITLE
Aesophia 430

### PR DESCRIPTION
**Pull request by radrow**
Apply Sophia 4.3.0 update and make use of some of its features

### CODE REVIEW

File: apps/aecontract/test/aect_test_utils.erl [hunk 1]
```diff
@@ -359,16 +359,26 @@ compile(Vsn, File) ->
...
-        {error, E} = Err -> io:format("~p\n", [E]), Err
+    ACIFlag = case aci_json_enabled(SophiaVsn) of
+                  yes_automatic -> [{aci, json}];
+                  _ -> []
+              end,
+    case aeso_compiler:from_string(Source, [{backend, fate}] ++ ACIFlag) of
+        {ok, Map} ->
+            case Map of
+                #{aci := ACI} ->
+                    AciId = make_aci_id(Source),
```
Comment (gorbak25) [inline apps/aecontract/test/aect_test_utils.erl:376]: You forgot to put this aci to my aci library

File: rebar.config [hunk 2]
```diff
@@ -211,7 +211,7 @@
                                  {sname, 'aeternity_ct@localhost'}]},
                     {deps, [{meck, "0.8.12"},
                             {websocket_client, {git, "git://github.com/aeternity/websocket_client", {ref, "a4fb3db"}}},
-                            {aesophia, {git, "https://github.com/aeternity/aesophia.git", {ref,"42cd47d"}}},
+                            {aesophia, {git, "https://github.com/aeternity/aesophia.git", {ref,"1266d9ea99dafcd12dfae001287f2b07e6d1e53e"}}},
```
Comment (gorbak25) [inline rebar.config:214]: Could you clip the reference to 5 chars

File: apps/aecontract/test/aect_test_utils.erl [hunk 3]
```diff
@@ -404,7 +415,13 @@ compiler_cmd(Vsn) ->
         ?SOPHIA_LIMA_FATE -> filename:join([BaseDir, "v4.3.1", "aesophia_cli"])
     end.
 
-aci_json_enabled(Vsn) -> Vsn > ?SOPHIA_ROMA andalso not aci_disabled().
+aci_json_enabled(Vsn) ->
+    case aci_disabled() of
+        true -> no;
+        _  when Vsn < ?SOPHIA_LIMA_FATE -> no;
```
Comment (gorbak25) [inline apps/aecontract/test/aect_test_utils.erl:421]: Only disable it for SOPHIA_ROMA - the aci lirary handles old compiles well.

**Review by gorbak25 — CHANGES_REQUESTED** [review]

File: rebar.lock [hunk 4]
```diff
@@ -160,5 +160,31 @@
  {<<"setup">>, <<"1203F4CDA11306C2E34434244576DED0A7BBFB0908D9A572356C809BD0CDF085">>},
  {<<"sext">>, <<"07D5D516DAA4C437B2F7DB99F6B2728C8AF939313479B1BEDC1995BB15936236">>},
  {<<"unicode_util_compat">>, <<"D869E4C68901DD9531385BB0C8C40444EBF624E60B6962D95952775CAC5E90CD">>},
- {<<"yamerl">>, <<"E51DBA652DCE74C20A88294130B48051EBBBB0BE7D76F22DE064F0F3CCF0AAF5">>}]}
+ {<<"yamerl">>, <<"E51DBA652DCE74C20A88294130B48051EBBBB0BE7D76F22DE064F0F3CCF0AAF5">>}]},
+{pkg_hash_ext,[
```
Comment (gorbak25) [inline rebar.lock:164]: Have you used a never rebar version than the one in the node?

**Review by gorbak25 — CHANGES_REQUESTED** [review]

File: rebar.lock (hunk 4 above)
Comment (radrow) [reply rebar.lock:164]: I am building only with `make` and `OTP 21`

File: apps/aecontract/test/aect_test_utils.erl (hunk 3 above)
Comment (radrow) [reply apps/aecontract/test/aect_test_utils.erl]: Should be fixed now

**Review by gorbak25 — APPROVED** [review]

File: test/contracts/protected_call.aes [hunk 5]
```diff
@@ -33,35 +33,41 @@ contract ProtectedCall =
   entrypoint get_balance() = Contract.balance
 
   payable stateful entrypoint
-
     id(n : int) = n
 
+  payable stateful entrypoint
```
Comment (hanssv) [inline test/contracts/protected_call.aes:38]: 😢

File: rebar.lock [hunk 6]
```diff
@@ -96,7 +96,7 @@
  {<<"leveled">>,
   {git,"https://github.com/martinsumner/leveled.git",
       {ref,"5bc137e4ef90d55f1e0da216a5ef5801d2a3d813"}},
-  0},
+  1},
```
Comment (hanssv) [inline rebar.lock:99]: I'm sure this digit is not too relevant?! But should we really change it?

**Review by hanssv — APPROVED** [review]

File: rebar.lock (hunk 6 above)
Comment (radrow) [reply rebar.lock:99]: Idk, it was autogenerated by shipped rebar

File: test/contracts/protected_call.aes (hunk 5 above)
Comment (radrow) [reply test/contracts/protected_call.aes:38]: bye c++, welcome java

File: rebar.lock (hunk 6 above)
Comment (hanssv) [reply rebar.lock:99]: Sounds like rebar, changing something without any reason 👍

**Pull request merged**